src/rust/dbv2/migrations/2024-01-17-184459_coingecko/up.sql

```diff
@@ -0,0 +1,192 @@
+    INNER JOIN api.coins quote
+    ON quote_account_address = quote."address" AND quote_module_name = quote.module AND quote_struct_name = quote.struct
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+CREATE OR REPLACE FUNCTION quote_indivisible_subunits_to_integer_price(market_id numeric, price numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT ($2 * POW(10,COALESCE(base.decimals, 0)) * lot_size) / tick_size
+    FROM market_registration_events
+    LEFT JOIN api.coins base
+    ON base_account_address = base."address" AND base_module_name = base.module AND base_struct_name = base.struct
+    INNER JOIN api.coins quote
+    ON quote_account_address = quote."address" AND quote_module_name = quote.module AND quote_struct_name = quote.struct
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+
+CREATE OR REPLACE FUNCTION size_to_base_indivisible_subunits(market_id numeric, "size" numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT $2 * lot_size
+    FROM market_registration_events
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+
+CREATE OR REPLACE FUNCTION size_and_price_to_quote_indivisible_subunits(market_id numeric, "size" numeric, price numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT $2 * $3 * tick_size
+    FROM market_registration_events
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+
+CREATE OR REPLACE FUNCTION get_quote_volume_divisor_for_market(numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT tick_size * POW(10::numeric,decimals::numeric)
+    FROM market_registration_events AS m
+    INNER JOIN api.coins AS c
+    ON m.quote_account_address = c."address"
+    AND m.quote_module_name = c.module
+    AND m.quote_struct_name = c.struct
+    WHERE market_id = $1;
+$$ LANGUAGE sql;
+
+
+-- Create utility functions used by the views
+CREATE FUNCTION integer_price_to_quote_nominal(market_id numeric, price numeric) RETURNS NUMERIC IMMUTABLE AS $$
+    SELECT integer_price_to_quote_indivisible_subunits(market_id, price) / get_quote_volume_divisor_for_market(market_id);
+$$ LANGUAGE sql;
+
+
+CREATE FUNCTION api.get_market_liquidity (market_id numeric, depth numeric) RETURNS NUMERIC AS $$
+    WITH mid_price AS (
+        SELECT
+          (MIN(price) FILTER (WHERE direction = 'ask') + MAX(price) FILTER (WHERE direction = 'bid')) / 2 AS mid_price
+        FROM api.orders
+        WHERE orders.market_id = $1
+        AND order_status = 'open'
+    )
+    SELECT SUM(CASE
+        WHEN direction = 'bid' THEN size_and_price_to_quote_indivisible_subunits($1,remaining_size,price)
+        ELSE size_and_price_to_quote_indivisible_subunits($1,remaining_size,mid_price)
+    END)
+    AS liquidity_in_quote
+    FROM api.orders, mid_price
+    WHERE order_status = 'open'
+    AND market_id = $1
+    AND price::numeric BETWEEN mid_price::numeric * (1::numeric - $2::numeric/10000::numeric) AND mid_price::numeric * (1::numeric + $2::numeric/10000::numeric);
```

Comment:
~~I think plus or minus 100% is actually overly opinionated here: e.g. a log function might be more appropriate since there is no upper bound on ask prices~~

~~If we adopt the liquidity pricing mechanism I propose above, I don't actually think we need to filter anything out~~

~~Because bids at a ridiculously low price tend to post negligible quote liquidity, and asks at a high price tend to post neglible base liquidity (when base liquidity is priced at last fill price)~~

~~And even at those ridiculous endpoints it is still technically liquidity that is up for grabs, and will fill during supply shocks~~

~~Since we've got a better pricing mechanism figured out, most liquidity will now be centered around the spread anyways, so I think we can actually get rid of this filter~~

Nevermind, I just saw that the filter at the end is for the +/- liqiduity calculations